logstash-rs/src/event.rs

```diff
@@ -68,6 +69,15 @@ impl LogStashRecord {
         self.fields.insert(key.into(), value);
         self
     }
+
+    pub fn with_data_from_map(mut self, extra_fields: &HashMap<String, Value>) -> Self {
+        self.fields.extend(
```

Comment:
I think, that checking that `extra_fields` is not empty before `extend` will save CPU a little bit, when there are no extra fields